packages/twenty-front/src/modules/ui/field/display/components/CurrencyDisplay.tsx

```diff
@@ -1,10 +1,22 @@
+import { useContext } from 'react';
+
+import { FieldContext } from '@/object-record/record-field/contexts/FieldContext';
+import { CurrencyDisplayWithIcons } from '@/ui/field/display/components/CurrencyDisplayWithIcon';
+
 import { EllipsisDisplay } from './EllipsisDisplay';
 
 type CurrencyDisplayProps = {
   amount?: number | null;
 };
 
-// TODO: convert currencyCode to currency symbol
 export const CurrencyDisplay = ({ amount }: CurrencyDisplayProps) => {
-  return <EllipsisDisplay>{amount}</EllipsisDisplay>;
+  const { showCurrencySymbol, currencyValues } = useContext(FieldContext);
```

Comment:
We already have `fieldValue` in `CurrencyFieldDisplay`, so you shouldn't have to pass these values in the `FieldContext`!

Here is what we could do:
- instead of `amount`, `CurrencyDisplay` should take only a `value` prop, with the type `FieldCurrencyValue` (most of the other field types use a "value" prop too).
- in `CurrencyFieldDisplay`:

```typescript
export const CurrencyFieldDisplay = () => {
  const { fieldValue } = useCurrencyField();

  return <CurrencyDisplay value={fieldValue} />;
};
```
- then in `CurrencyDisplay`, we can compute the needed values like this:
```typescript
const amount = value?.amountMicros ? value.amountMicros / 1000000 : null;
const currencyCode = value?.currencyCode ?? CurrencyCode.USD;
```

This way you don't need to pass those values in the context!

Reply:
hi @thaisguigon `fieldValue.currencyCode` value is always USD in `CurrencyFieldDisplay` (This was the primary reason why I go with using context here), and I am failing to understand how the `currencyCode` should be updated in useCurrencyField when we select a new unit from the selection. Appreciate your help here :)

Reply:
Fixed.

@anoopw3bdev We want to extract from the field value with useCurrencyField and then fieldValue, then drill it down to CurrencyDisplay which is stateless.